src/FilenameParsing/FileIDHelperResolutionStrategy.php

```diff
@@ -255,7 +257,14 @@ private function validateHash(FileIDHelper $helper, ParsedFileID $parsedFileID,
 
         // Check if the physical hash of the file starts with our parsed file ID hash
         $actualHash = $this->findHashOf($helper, $parsedFileID, $filesystem);
-        return strpos($actualHash, $parsedFileID->getHash()) === 0;
+        if (!$actualHash) {
+            return false;
+        }
+
+        /** @var FileHashingService $hasher */
+        $hasher = Injector::inst()->get(FileHashingService::class);
```

Comment:
I'd rater make it a dependency on the strategy. I could imagine future scenario where other classes would use this. Also, the framework PR need an reference to it as well, so it can enable caching.

The ShortCodeTask should enable it as well

Reply:
Good point about shortcodes - so it's effectively a global service. In terms of OOP, it's still a bit messy (global injection points). But it's not a commonly customised API, just an internal detail that will quite likely only ever have one implementation - so I'm not too fussed.

Reply:
Done. I made that dependency injectable.